src/IssueViz.Security/OpenInIDE/Api/StatusRequestHandler.cs

```diff
@@ -31,22 +31,34 @@ namespace SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Api
     [Export(typeof(IStatusRequestHandler))]
     internal class StatusRequestHandler : IStatusRequestHandler
     {
-        private readonly IVsShell vsShell;
-        private readonly IVsSolution vsSolution;
+        private readonly IVsUIServiceOperation vSServiceOperation;
 
         [ImportingConstructor]
-        public StatusRequestHandler([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        public StatusRequestHandler(IVsUIServiceOperation vSServiceOperation)
         {
-            vsShell = serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
-            vsSolution = serviceProvider.GetService(typeof(SVsSolution)) as IVsSolution;
+            this.vSServiceOperation = vSServiceOperation;
         }
 
         Task<IStatusResponse> IStatusRequestHandler.GetStatusAsync()
         {
-            vsShell.GetProperty((int) __VSSPROPID5.VSSPROPID_AppBrandName, out var ideName);
+            var result = vSServiceOperation.Execute<SVsShell, IVsShell, Task<IStatusResponse>>(
+                shell =>
+                {
+                    return vSServiceOperation.Execute<SVsSolution, IVsSolution, Task<IStatusResponse>>(solution =>
+                    {
+                        return DoGetStatusAsync(shell, solution);
+                    });
+                });
+
+            return result;
+        }
+
+        private Task<IStatusResponse> DoGetStatusAsync(IVsShell shell, IVsSolution solution)
+        {
+            shell.GetProperty((int)__VSSPROPID5.VSSPROPID_AppBrandName, out var ideName);
             ideName = ideName ?? "Microsoft Visual Studio";
 
-            vsSolution.GetProperty((int) __VSPROPID.VSPROPID_SolutionBaseName, out var ideInstance);
+            solution.GetProperty((int)__VSPROPID.VSPROPID_SolutionBaseName, out var ideInstance);
             ideInstance = ideInstance ?? "";
 
             return Task.FromResult<IStatusResponse>(new StatusResponse(ideName.ToString(), ideInstance.ToString()));
```

Comment:
There's something a bit weird about this; this method returns a `Task` and service operation has an `ExecuteAsync` method, so it shouldn't be necessary to use `Task.FromResult<...>`.